packages/lodestar/src/chain/clock/LocalClock.ts

```diff
@@ -0,0 +1,69 @@
+import {Epoch, Slot} from "@chainsafe/lodestar-types";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {computeEpochAtSlot, getCurrentSlot} from "@chainsafe/lodestar-beacon-state-transition";
+
+import {ChainEventEmitter} from "../emitter";
+
+import {IBeaconClock} from "./interface";
+
+/**
+ * A local clock, the clock time is assumed to be trusted
+ */
+export class LocalClock implements IBeaconClock {
+  private readonly config: IBeaconConfig;
+  private readonly genesisTime: number;
+  private timeoutId: NodeJS.Timeout;
+  private readonly signal: AbortSignal;
+  private readonly emitter: ChainEventEmitter;
+  private _currentSlot: number;
+
+  public constructor({
+    config,
+    genesisTime,
+    emitter,
+    signal,
+  }: {
+    config: IBeaconConfig;
+    genesisTime: number;
+    emitter: ChainEventEmitter;
+    signal: AbortSignal;
+  }) {
+    this.config = config;
+    this.genesisTime = genesisTime;
+    this.timeoutId = setTimeout(this.onNextSlot, this.msUntilNextSlot());
+    this.signal = signal;
+    this.emitter = emitter;
+    this._currentSlot = getCurrentSlot(this.config, this.genesisTime);
+    signal.addEventListener("abort", this.abort);
+  }
+
+  public get currentSlot(): Slot {
+    return this._currentSlot;
+  }
+
+  public get currentEpoch(): Epoch {
+    return computeEpochAtSlot(this.config, this._currentSlot);
+  }
+
+  private abort = (): void => {
+    clearTimeout(this.timeoutId);
```

Comment:
if(this.timeoutId) {
}

Reply:
We don't need that anymore with the clock refactoring, `this.timeout` is always set.

Reply:
what about calling this method twice in a row?